src/templates/gwfcore/gwfcore-launch-template.template.yaml

```diff
@@ -71,126 +77,145 @@ Resources:
+              - unzip -q /tmp/awscliv2.zip -d /tmp
+              - /tmp/aws/install -b /usr/bin
+
+              # check that the aws-cli was actually installed. if not shutdown (terminate) the instance
+              - command -v aws || shutdown -P now
+
+              - mkdir -p /opt/aws-cli/bin
+              - cp -a $(dirname $(find /usr/local/aws-cli -name 'aws' -type f))/. /opt/aws-cli/bin/
 
-                runcmd:
+              # set environment variables for provisioning
+              - export GWFCORE_NAMESPACE=${Namespace}
+              - export INSTALLED_ARTIFACTS_S3_ROOT_URL=$(aws ssm get-parameter --name /gwfcore/${Namespace}/installed-artifacts/s3-root-url --query 'Parameter.Value' --output text)
 
-                # start the amazon-cloudwatch-agent
-                - /opt/aws/amazon-cloudwatch-agent/bin/amazon-cloudwatch-agent-ctl -a fetch-config -m ec2 -s -c file:/opt/aws/amazon-cloudwatch-agent/etc/config.json
+              # enable ecs spot instance draining
+              - echo ECS_ENABLE_SPOT_INSTANCE_DRAINING=true >> /etc/ecs/ecs.config
 
-                # install aws-cli v2 and copy the static binary in an easy to find location for bind-mounts into containers
-                - curl -s "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "/tmp/awscliv2.zip"
-                - unzip -q /tmp/awscliv2.zip -d /tmp
-                - /tmp/aws/install -b /usr/bin
+              # pull docker images only if missing
+              - echo ECS_IMAGE_PULL_BEHAVIOR=prefer-cached >> /etc/ecs/ecs.config
 
-                # check that the aws-cli was actually installed. if not shutdown (terminate) the instance
-                - command -v aws || shutdown -P now
+              - cd /opt
+              - aws s3 sync $INSTALLED_ARTIFACTS_S3_ROOT_URL/ecs-additions ./ecs-additions
+              - chmod a+x /opt/ecs-additions/provision.sh
+              - /opt/ecs-additions/provision.sh
 
-                - mkdir -p /opt/aws-cli/bin
-                - cp -a $(dirname $(find /usr/local/aws-cli -name 'aws' -type f))/. /opt/aws-cli/bin/
+              # optional EFS mount
+              ${EfsMount}
 
-                # set environment variables for provisioning
-                - export GWFCORE_NAMESPACE=${Namespace}
-                - export INSTALLED_ARTIFACTS_S3_ROOT_URL=$(aws ssm get-parameter --name /gwfcore/${Namespace}/installed-artifacts/s3-root-url --query 'Parameter.Value' --output text)
+              --==BOUNDARY==--
 
-                - cd /opt
-                - aws s3 sync $INSTALLED_ARTIFACTS_S3_ROOT_URL/ecs-additions ./ecs-additions
-                - chmod a+x /opt/ecs-additions/provision.sh
-                - /opt/ecs-additions/provision.sh
 
-                --==BOUNDARY==--
+            - EfsMount: !If
+              - NoEfs
+              - ""
+              - !Sub |
+                - mkdir -p /mnt/efs
+                - echo "${EfsId}:/ /mnt/efs efs tls,_netdev" >> /etc/fstab
```

Comment:
The EFS ID is stored in an SSM parameter, can we use that here instead of hard coding the value in the launch template?

Reply:
good point. will address

Reply:
actually, no we can't in current form. If you use an existing EFS ID, there is no guarantee it will be an SSM parameter. We *could probably* fix this with the appropriate `DependsOn` ordering in `gwfcore-root` where we put in the `ExistingEFS` parameter into SSM namespace. Thoughts?

Reply:
Not sure I understand why the SSM parameter wouldn't (or couldn't) be guaranteed. If using an existing EFS ID, couldn't that ID be passed as the value to the SSM parameter?